src/aux-vm-browser/managers/LivekitManager.ts

```diff
@@ -98,9 +98,24 @@ export class LivekitManager implements SubscriptionLike {
         this._addressToTrack = null;
         this._trackToAddress = null;
     }
+    private async _requestMediaPermissions(): Promise<void> {
+        // Request audio and video permission (required)
+        try {
+            await navigator.mediaDevices.getUserMedia({
+                video: true,
+                audio: true,
+            });
+            console.log('[LivekitManager] Media permissions granted.');
+        } catch (error) {
+            console.error('[LivekitManager] Media permissions denied:', error);
+            throw new Error('Media permissions are required to join the room.');
+        }
+    }
```

Comment:
@TroyceGowdy Do you know if this error is able to be handled by the CasualOS script that called `os.joinRoom()`?

Reply:
Ok, but would the script be able to get the error and automatically check that the user didn't grant permission?

Reply:
Yes it does. Just checked the toast message and it is coming through.

Reply:
Ok, good. Also one more question: if the user has already granted permissions, does it prompt for permissions again or does it just join the room?

Reply:
It just joins the room